src/packages/LiveClient.ts

```diff
@@ -16,33 +16,62 @@ import type {
 } from "../lib/types";
 
 export class LiveClient extends AbstractWsClient {
-  private _socket: w3cwebsocket;
+  protected _socket: w3cwebsocket;
+  protected _reconnections: number = 0;
 
   constructor(
     protected key: string,
     protected options: DeepgramClientOptions | undefined = DEFAULT_OPTIONS,
     private transcriptionOptions: LiveSchema = {},
-    endpoint = "v1/listen"
+    protected endpoint = "v1/listen"
   ) {
     super(key, options);
 
-    const url = new URL(endpoint, this.baseUrl);
-    url.protocol = url.protocol.toLowerCase().replace(/(http)(s)?/gi, "ws$2");
-    appendSearchParams(url.searchParams, this.transcriptionOptions);
+    this._socket = this._connect();
+    this._registerEvents();
+  }
+
+  protected _connect(): w3cwebsocket {
+    const url = buildRequestUrl(this.endpoint, this.baseUrl, this.transcriptionOptions);
+
+    try {
+      const socket = new w3cwebsocket(url.toString(), ["token", this.key]);
+
+      return socket;
+    } catch (error: any) {
+      throw new DeepgramError(error);
+    }
+  }
 
-    this._socket = new w3cwebsocket(url.toString(), ["token", this.key]);
+  protected _attemptReconnection(): void {
+    if (this._reconnections > WS_RECONNECTION_RETRY_LIMIT) {
+      throw new DeepgramError("Reconnection retry limit exceeded");
+    }
+
+    this._reconnections += 1;
+    this._unregisterEvents();
```

Comment:
Would the user be interested in getting the `onopen` event to know that the WS successfully reconnected?

Reply:
We reregister the events after. These internal events emit an external one, which won't need to be reregistered. So they're likely to get second onopen event... which may be troublesome for some actually